pkg/core/system.go

```diff
@@ -164,24 +204,79 @@ To fix this you need to:
  3. Re-install ` + env.Cli.Name + `
 
 `)
-			return false, err // hard error
 		}
-		if willRetry {
-			return false, nil // retriable error
+		return err
+	}
+	return nil
+}
+
+// TODO this only supports checking Pods for phases
+func (c *client) checkResource(resource crd.RiffResources) error {
+	cnt := 1
+	for _, check := range resource.Checks {
+		var ready bool
+		var err error
+		for i := 0; i< 360; i++ {
+			if strings.EqualFold(check.Kind, "Pod") {
+				ready, err = c.isPodReady(check, resource.Namespace)
+				if err != nil {
+					return err
+				}
+				if ready {
+					break
+				}
+			} else {
+				return errors.New("only Kind:Pod supported for resource checks")
+			}
+			time.Sleep(1 * time.Second)
+			cnt++
+			if cnt % 5 == 0 {
+				fmt.Print(".")
+			}
+		}
+		if !ready {
+			return errors.New(fmt.Sprintf("The resource %s did not initialize", resource.Name))
 		}
-		return false, err // not retrying, so treat as hard error
 	}
-	return true, nil // success
+	fmt.Println("done")
+	return nil
+}
+
+func (c *client) isPodReady(check crd.ResourceChecks, namespace string) (bool, error) {
+	pods := c.kubeClient.CoreV1().Pods(namespace)
+	podList, err := pods.List(metav1.ListOptions{
+		LabelSelector: convertMapToString(check.Selector.MatchLabels),
+	})
+	if err != nil {
+		return false, err
+	}
+	for _, pod := range podList.Items {
+		if strings.EqualFold(string(pod.Status.Phase), check.Pattern) {
+			return true, nil
+		}
+	}
+	return false, nil
 }
 
-func (kc *kubectlClient) SystemUninstall(options SystemUninstallOptions) (bool, error) {
+func convertMapToString(m map[string]string) string {
+	var s string
+	for k,v := range m {
+		s += k + "=" + v + ","
+	}
+	if last := len(s) - 1; last >= 0 && s[last] == ',' {
+		s = s[:last]
+	}
+	return s
+}
+
+func (c *client) SystemUninstall(options SystemUninstallOptions) (bool, error) {
```

Comment:
these changes should be pulled into a separate PR. The Manifest CRD has no direct impact on uninstall. At least not in its current incarnation.

Reply:
cool!